asyncapi/document.go

```diff
@@ -80,14 +80,22 @@ type Message interface {
 	Payload() Schema
 }
 
+// FalsifiableSchema is a variadic type used for some Schema fields.
+// For example, additionalProperties value can be either `false` or a Schema.
+type FalsifiableSchema interface {
+	IsFalse() bool
+	IsSchema() bool
+	Schema() Schema
+}
```

Comment:
Sorry if I'm being disruptive with this review 😅   I think that `FalsifiableSchema` isn't correct solution. Of course, `additionalProperties` and `additionalItems` have option to be `false`, but also have option to be `true` :) Also, the JSON Schema can be either a boolean or Schema object (in places where you can use the Schema, like in `items` or `properties`), so I think that:

```go
type BooleanbleSchema interface {
        isBoolean() bool
	Boolean() bool
        IsSchema() bool
	Schema() Schema
}
```

should be better to handle Schema, true and false :) - like we have in our ParserJS. https://github.com/asyncapi/parser-js/blob/master/lib/models/schema.js#L368

Reply:
No you are not! This is the kind of review I'm looking for. You have a lot of experience with JSON Schema, a thing I have not :) 
I think we can move forward with this solution on this PR. I opened an issue here https://github.com/asyncapi/event-gateway/issues/22 which I'll work next on it for improving the whole Schema thing, including this boolean behavior you mentioned here.

Reply:
Ok :)